sensor/common/scan/scan.go

```diff
@@ -68,22 +74,22 @@ func NewLocalScan(registryStore *registry.Store) *LocalScan {
 }
 
 // EnrichLocalImageInNamespace invokes enrichLocalImageFromRegistry with a slice of credentials from the registryStore based on namespace as well as
-// the OCP global pull secret
+// the OCP global pull secret.
 //
-// If no registry credentials are found an empty registry slice is passed to enrichLocalImageFromRegistry for enriching with 'no auth'
-func (s *LocalScan) EnrichLocalImageInNamespace(ctx context.Context, centralClient v1.ImageServiceClient, ci *storage.ContainerImage, namespace string) (*storage.Image, error) {
-	if namespace == "" {
-		// If no namespace provided try enrichment with 'no auth'
-		return s.enrichLocalImageFromRegistry(ctx, centralClient, ci, nil)
-	}
-
+// If no registry credentials are found an empty registry slice is passed to enrichLocalImageFromRegistry for enriching with 'no auth'.
+func (s *LocalScan) EnrichLocalImageInNamespace(ctx context.Context, centralClient v1.ImageServiceClient, ci *storage.ContainerImage, namespace string, requestID string, force bool) (*storage.Image, error) {
```

Comment:
Something for the future, but not in this PR: this might be an opportunity for the options pattern again, seeing as this will only be optionally set by a few specific callers.

Reply:
Agreed! I felt dirty adding another flag.  Part of me wants to re-write all the enrichment logic, the differences/edge cases in central, sensor, ad-hoc, not, and all the various flags / contexts is (IMHO) overly complex.